pinot-spi/src/main/java/org/apache/pinot/spi/stream/StreamMetadataProvider.java

```diff
@@ -53,4 +57,34 @@ default StreamPartitionMsgOffset fetchStreamPartitionOffset(@Nonnull OffsetCrite
     long offset = fetchPartitionOffset(offsetCriteria, timeoutMillis);
     return new LongMsgOffset(offset);
   }
+
+  /**
+   * Fetch the list of partition group info for the latest state of the stream.
+   * Default behavior is the one for the Kafka stream, where each partition group contains only one partition
+   * @param currentPartitionGroupsMetadata The list of metadata for the current partition groups
+   */
+  default List<PartitionGroupInfo> getPartitionGroupInfoList(String clientId, StreamConfig streamConfig,
+      List<PartitionGroupMetadata> currentPartitionGroupsMetadata, int timeoutMillis)
+      throws TimeoutException, IOException {
+    int partitionCount = fetchPartitionCount(timeoutMillis);
+    List<PartitionGroupInfo> newPartitionGroupInfoList = new ArrayList<>(partitionCount);
+
+    // Add a PartitionGroupInfo into the list foreach partition already present in current.
+    for (PartitionGroupMetadata currentPartitionGroupMetadata : currentPartitionGroupsMetadata) {
+      newPartitionGroupInfoList.add(new PartitionGroupInfo(currentPartitionGroupMetadata.getPartitionGroupId(),
+          currentPartitionGroupMetadata.getEndOffset()));
```

Comment:
It looks that we are setting `endOffset` to be the `startOffset` of the new partition group. 

Does this mean either `startOffset` or `endOffset` is exclusive? If so, please add the documentation accordingly in `PartitionGroupMetadata`

Reply:
Done. As far as the current impl is concerned, that offset is never going to be used.